flux/stdlib/influxdata/influxdb/schema/read_tag_keys_test.flux

```diff
@@ -0,0 +1,157 @@
+,,3,2018-05-22T19:54:06Z,system,us-east,host.local,load1,17
+,,3,2018-05-22T19:54:16Z,system,us-east,host.local,load1,17
+
+,,4,2018-05-22T19:53:26Z,system,us-east,host.local,load3,16
+,,4,2018-05-22T19:53:36Z,system,us-east,host.local,load3,16
+,,4,2018-05-22T19:53:46Z,system,us-east,host.local,load3,15
+,,4,2018-05-22T19:53:56Z,system,us-east,host.local,load3,19
+,,4,2018-05-22T19:54:06Z,system,us-east,host.local,load3,19
+,,4,2018-05-22T19:54:16Z,system,us-east,host.local,load3,19
+
+,,5,2018-05-22T19:53:26Z,system,us-west,host.local,load5,19
+,,5,2018-05-22T19:53:36Z,system,us-west,host.local,load5,22
+,,5,2018-05-22T19:53:46Z,system,us-west,host.local,load5,11
+,,5,2018-05-22T19:53:56Z,system,us-west,host.local,load5,12
+,,5,2018-05-22T19:54:06Z,system,us-west,host.local,load5,13
+,,5,2018-05-22T19:54:16Z,system,us-west,host.local,load5,13
+"
+
+    want = csv.from(
+        csv: "
+#datatype,string,long,string
+#group,false,false,false
+#default,0,,
+,result,table,_value
+,,0,_field
+,,0,_measurement
+,,0,_start
+,,0,_stop
+,,0,host
+,,0,region
+",
+    )
+    result = testing.loadStorage(csv: input)
+        |> range(start: -100y)
+        |> filter(fn: (r) => true)
+        |> keys()
+        |> keep(columns: ["_value"])
+        |> distinct()
+        |> sort()
+    testing.diff(want: want, got: result)
+}
+
+testcase read_tag_keys_with_predicate {
+    expect.planner(rules: ["PushDownReadTagKeysRule": 1])
+
+    input = "
+#group,false,false,false,false,true,true,true,true,true,true,true
+#datatype,string,long,dateTime:RFC3339,long,string,string,string,string,string,string,string
+#default,_result,,,,,,,,,,
+,result,table,_time,_value,_field,_measurement,device,fstype,host,mode,path
+,,0,2020-10-21T20:48:30Z,4881964326,inodes_free,disk,disk1s5,apfs,euterpe.local,ro,/
+,,0,2020-10-21T20:48:40Z,4881964326,inodes_free,disk,disk1s5,apfs,euterpe.local,ro,/
+,,0,2020-10-21T20:48:50Z,4881964326,inodes_free,disk,disk1s5,apfs,euterpe.local,ro,/
+,,1,2020-10-21T20:48:30Z,4294963701,inodes_free,disk,disk2s1,hfs,euterpe.local,ro,/Volumes/IntelliJ IDEA CE
+,,1,2020-10-21T20:48:40Z,4294963701,inodes_free,disk,disk2s1,hfs,euterpe.local,ro,/Volumes/IntelliJ IDEA CE
+,,1,2020-10-21T20:48:50Z,4294963701,inodes_free,disk,disk2s1,hfs,euterpe.local,ro,/Volumes/IntelliJ IDEA CE
+,,2,2020-10-21T20:48:30Z,488514,inodes_used,disk,disk1s5,apfs,euterpe.local,ro,/
+,,2,2020-10-21T20:48:40Z,488514,inodes_used,disk,disk1s5,apfs,euterpe.local,ro,/
+,,2,2020-10-21T20:48:50Z,488514,inodes_used,disk,disk1s5,apfs,euterpe.local,ro,/
+,,3,2020-10-21T20:48:30Z,3578,inodes_used,disk,disk2s1,hfs,euterpe.local,ro,/Volumes/IntelliJ IDEA CE
+,,3,2020-10-21T20:48:40Z,3578,inodes_used,disk,disk2s1,hfs,euterpe.local,ro,/Volumes/IntelliJ IDEA CE
+,,3,2020-10-21T20:48:50Z,3578,inodes_used,disk,disk2s1,hfs,euterpe.local,ro,/Volumes/IntelliJ IDEA CE
+
+#group,false,false,false,false,true,true,true,true
+#datatype,string,long,dateTime:RFC3339,double,string,string,string,string
```

Comment:
It would be better if there was another tag key here to prove that it was filtered out